modules/hdf5_handler/h5get.cc

```diff
@@ -981,38 +981,58 @@ string print_attr(hid_t type, int loc, void *sm_buf) {
             if (H5Tget_size(type) == 4) {
                 
                 float attr_val = *(float*)sm_buf;
-                bool is_a_fin = isfinite(attr_val);
-                // Represent the float number.
-                // Some space may be wasted. But it is okay.
-                gp.tfp = (float *) sm_buf;
-                int ll = snprintf(gps, 30, "%.10g", *(gp.tfp + loc));
+                // Note: this comparsion is the same as isnan.
+                // However, on CentOS 7, isnan() is declared in two headers and causes conflicts.
+                if (attr_val!=attr_val) {
+                    rep.resize(3);
+                    rep[0]='N';
```

Comment:
You could use something like 'strncpy(rep.data(), "NaN", sizeof(NaN));' but it's probably 6 to one and half a dozen to the other. But... are you going to expect a null character at the end? The snprintf() function does append nulls.

Reply:
This again is the original code. I don't want to spend time changing the original code with this ticket. The null character doesn't show up in the final DMR and fileout netCDF output. So let's keep this code.